pkg/api.go

```diff
@@ -234,34 +234,22 @@ func (a *API) specificRegistration(w http.ResponseWriter, r *http.Request) (int,
 
 func (a *API) proposerPayloadsDelivered(w http.ResponseWriter, r *http.Request) (int, error) {
 	slot, err := specificSlot(r)
-	if err == nil {
-		blocks, err := a.Service.GetDelivered(r.Context(), slot)
-		return a.respond(w, blocks, err)
-	} else if err != nil && !errors.Is(err, ErrParamNotFound) {
+	if err != nil && !errors.Is(err, ErrParamNotFound) {
```

Comment:
Yeah, so basically, checks whether the parameter exists. If not it raises the `ErrParamNotFound` and returns the zero value for the parameter. For example, it may not contain the slot parameter, so it will return the error and `Slot(0)`. We could use a boolean instead, but then, we would not catch other errors like parsing.